examples/rosettacode/doors_with_classes.nit

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env nit
+#
+# This file is part of NIT ( http://www.nitlanguage.org ).
+# This program is public domain
+
+# Task: 100 doors
+# SEE: <http://rosettacode.org/wiki/100_doors>
+
+# A door with two states: open or closed
+class Door
+	# Is this door open?
+	var open = false
+
+	# Toggle bool value of open
+	fun toggle do open = not open
+
+	redef fun to_s: String
+	do
+		return if open then "Open" else "Closed"
+	end
+end
+
+var doors = new Array[Door]
+
+for door in [0..100[ do doors.add(new Door)
+
+var i = 0
+var visit = 1
+
+while i < 100 do
+	doors[i].toggle
+
+	i += visit
+
+	if i >= 100 then
+		if visit == 100 then
+			print "Porte {visit} : {doors[visit - 1]}"
+		else
+			i = visit
+		end
+
+		visit += 1
+	end
+end
```

Comment:
Tes boucles et if sont trop compliqués. Simplifie:

~~~nit
for visit in [0..n[ do
        var i = visit
        while i < n do
                doors[i].toggle
                i += visit+1
        end
end
for i in [0..n[ do print "Porte {i} : {doors[i]}
~~~